lib/collection/response.js

```diff
@@ -107,6 +107,8 @@ _.inherit((
 
 _.assign(Response.prototype, /** @lends Response.prototype */ {
     update: function (options) {
+        // options.stream accepts new Buffer() as well as new Buffer().toJSON()
+        options.stream && options.stream.type === 'Buffer' && (options.stream = new Buffer(options.stream));
```

Comment:
1. The change foes inside mergeDefined (L144)
2. In the turnery else block, detect type=Buffer && data=Array then
3. If Buffer present, create Buffer
4. else create ArrayBuffer from data Array (keep performance in mind)